### DISCUSSION TITLE
How to add support for a new tablet

### DISCUSSION THREAD

**Discussion by Randommist** [OpenTabletDriver/OpenTabletDriver#3213] [Q&A] [answered]
Hello.  I became the owner of a Huion inspiroy 2 m (H951P) graphics tablet and I am not very happy with the proprietary driver offered by Huion.  I would like to help add support for this tablet to the driver.  Having read the source code a little, I came to the conclusion that in order for you to support you need to create a configuration file.  Can you describe how the process of adding support for a new tablet occurs and where to get information for the configuration file?

**Comment by jamesbt365** [comment, marked as answer]
There is no documentation for creating configurations at the moment and the the process isn't exactly very streamlined to begin with, however, a configuration already exists for this tablet and its pending release into a new version, see #2927.

You may use this file, but on `0.6.x` pen buttons are defined differently. So you need to change the pen button definition to this:
```
    "Pen": {
      "MaxPressure": 8191,
      "Buttons": {
        "ButtonCount": 2
      }
    },
```

You should also replace the `ReportParser` with `"OpenTabletDriver.Configurations.Parsers.UCLogic.UCLogicTiltReportParser`.

just replace it with the "code" above and place the finished file in in `%localappdata%\OpenTabletDriver\Configurations` for Windows and `~/.local/share/OpenTabletDriver/Configurations` for Linux.

Its worth noting that the group keys will work as regular auxiliary buttons, but if the configuration doesn't work well at all you will either need to wait for the next release of OpenTabletDriver, backport the changes yourself or use a development build which is very experimental and is unstable.
